nixos/modules/services/desktops/gnome3/gnome-settings-daemon.nix

```diff
@@ -40,6 +40,18 @@ in
 
     services.udev.packages = [ cfg.package ];
 
+    security.wrappers.gsd-backlight-helper.source = "${pkgs.gnome3.gnome-settings-daemon}/libexec/gsd-backlight-helper";
+
+    nixpkgs.overlays = [
+      (self: super: {
+        gnome3 = super.gnome3.overrideScope' (gself: gsuper: {
+          gnome-settings-daemon = gsuper.gnome-settings-daemon.override {
+            inherit (config.security) wrapperDir;
```

Comment:
Do we really gain much by passing `config.security.wrapperDir` through an overlay btw? It's not supposed to be modified, so the only reason to do the override is not having to change the default in `gnome-settings-daemon` if the wrapperDir changes (which I'm guessing should happen very rarely). Perhaps we could use https://github.com/NixOS/nixpkgs/pull/56227 when that lands instead?

Or am I missing something here?

Reply:
Yeah, I missed it is not supposed to be modified. We can drop this.

Reply:
> Do we really gain much by passing config.security.wrapperDir through an overlay btw? It's not supposed to be modified, so the only reason to do the override is not having to change the default in gnome-settings-daemon if the wrapperDir changes (which I'm guessing should happen very rarely)

No disagreement on that point, description of `wrapperDir` says:
> It should not be overriden.

Though one concern from the original pr that maybe needs some care
> I have this vague feeling that the entire preFixup hook should only happen when the NixOS module is what's using the gsd package, since otherwise there's no /run/wrappers copy of the helper.

Reply:
> I have this vague feeling that the entire preFixup hook should only happen when the NixOS module is what's using the gsd package, since otherwise there's no /run/wrappers copy of the helper.

Yeah, the backlight stuff won't probably work without the module, so it's not super pretty, but probably okay since it's unlikely that anyone uses it without the module on NixOS. On non-NixOS I would guess that this is a package that's simply not used?

Reply:
Outside of NixOS, they previously still needed to replicate polkit module for the backlight helper to work. And systemd module if they wanted to actually run g-s-d.